src/main/java/com/laytonsmith/core/events/EventUtils.java

```diff
@@ -181,12 +181,105 @@ public static void ManualTrigger(String eventName, CArray object, Target t, bool
 		}
 	}
 
+	/**
+	 * This returns whether or not the given BoundEvent's prefilters matches the event. This method will never
+	 * throw a {@link PrefilterNonMatchException}
+	 * @param b
+	 * @param e
+	 * @param driver
+	 * @return
+	 */
+	public static boolean PrefilterMatches(BoundEvent b, BindableEvent e, Event driver) {
+		Event.PrefilterBuilder prefilterBuilder = driver.getPrefilters();
+		Map<String, Mixed> userPrefilters = b.getPrefilter();
+		if(prefilterBuilder == null) {
+			// Old, deprecated method
+			try {
+				return driver.matches(userPrefilters, e);
+			} catch (PrefilterNonMatchException ex) {
+				return false;
+			}
+		} else {
+			Map<String, Pair<Prefilters.PrefilterType, Event.PrefilterMatcher>> prefilters = prefilterBuilder.build();
+			for(Map.Entry<String, Mixed> prefilter : userPrefilters.entrySet()) {
+				if(!prefilters.containsKey(prefilter.getKey())) {
+					// The compiler should have already warned about this
+					continue;
+				}
+				Pair<Prefilters.PrefilterType, Event.PrefilterMatcher> pair = prefilters.get(prefilter.getKey());
+				Event.PrefilterMatcher matcher = pair.getValue();
+				Mixed value = prefilter.getValue();
+				switch(pair.getKey()) {
+					case STRING_MATCH: {
+						String compare = matcher.getStringParameter(e);
+						if(compare != null && !Prefilters.FastStringMatch(compare, value.val())) {
```

Comment:
This means there's no way to handle case sensitivity in event prefilters using this new method. STRING_IC_MATCH?

Reply:
There's also no way to handle nulls.

Reply:
Hmm, don't think that was possible already, I'm pretty sure I kept the same behavior as before. https://github.com/EngineHub/CommandHelper/blob/master/src/main/java/com/laytonsmith/core/events/Prefilters.java#L169 Or are you suggesting a new feature?